src/PrestaShopBundle/Security/Annotation/DemoRestricted.php

```diff
@@ -63,6 +60,18 @@ class DemoRestricted extends ConfigurationAnnotation
      */
     protected $redirectQueryParamsToKeep = [];
 
+    public function __construct(
+        ?string $redirectRoute = null,
+        string $domain = 'Admin.Notifications.Error',
+        string $message = 'This functionality has been disabled.',
+        array $redirectQueryParamsToKeep = []
+    ) {
+        $this->domain = $domain;
+        $this->message = $message;
+        $this->redirectRoute = $redirectRoute;
+        $this->redirectQueryParamsToKeep = $redirectQueryParamsToKeep;
+    }
```

Comment:
Do you think that we can use constructor property promotion on this?

Reply:
Sure, I updated the code